networkx/utils/misc.py

```diff
@@ -175,3 +177,16 @@ def dict_to_numpy_array1(d,mapping=None):
         i = mapping[k1]
         a[i] = d[k1]
     return a
+
+# Recipe from the itertools documentation.
+def consume(iterator):
```

Comment:
consume is what I needed in the algorithm.  What's the count recipe?  `sum(1 for _ in iterator)`?